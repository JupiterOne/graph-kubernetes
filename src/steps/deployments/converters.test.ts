import {
  createContainerSpecEntity,
  createContainerSpecToVolumeRelationship,
  createDeploymentEntity,
  createVolumeEntity,
} from './converters';
import {
  createMockContainer,
  createMockDeployment,
  createMockVolume,
  createMockVolumeMount,
} from '../../../test/mocks';

describe('#createDeploymentEntity', () => {
  test('should convert data', () => {
    expect(createDeploymentEntity(createMockDeployment())).toMatchSnapshot();
  });
});

describe('#createContainerSpecToVolumeRelationship', () => {
  test('should generate relationship with mountPath', () => {
    const namespaceName = 'default';
    const deploymentId = '12345';

    expect(
      createContainerSpecToVolumeRelationship({
        containerSpecEntity: createContainerSpecEntity(
          namespaceName,
          deploymentId,
          createMockContainer(),
        ),
        volumeEntity: createVolumeEntity(
          namespaceName,
          deploymentId,
          createMockVolume(),
        ),
        volumeMount: createMockVolumeMount(),
      }),
    ).toMatchSnapshot();
  });

  test('should generate relationship without mountPath', () => {
    const namespaceName = 'default';
    const deploymentId = '12345';

    expect(
      createContainerSpecToVolumeRelationship({
        containerSpecEntity: createContainerSpecEntity(
          namespaceName,
          deploymentId,
          createMockContainer(),
        ),
        volumeEntity: createVolumeEntity(
          namespaceName,
          deploymentId,
          createMockVolume(),
        ),
        volumeMount: createMockVolumeMount({
          mountPath: undefined,
        }),
      }),
    ).toMatchSnapshot();
  });

  test('should generate relationship with mountPath and subPath', () => {
    const namespaceName = 'default';
    const deploymentId = '12345';

    expect(
      createContainerSpecToVolumeRelationship({
        containerSpecEntity: createContainerSpecEntity(
          namespaceName,
          deploymentId,
          createMockContainer(),
        ),
        volumeEntity: createVolumeEntity(
          namespaceName,
          deploymentId,
          createMockVolume(),
        ),
        volumeMount: createMockVolumeMount({
          subPath: 'abc',
        }),
      }),
    ).toMatchSnapshot();
  });
});
