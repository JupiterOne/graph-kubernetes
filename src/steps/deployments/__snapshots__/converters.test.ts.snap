// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createContainerSpecToVolumeRelationship should generate relationship with mountPath 1`] = `
{
  "_class": "USES",
  "_fromEntityKey": "12345/coredns",
  "_key": "12345/coredns|uses|default/12345/config-volume/mount_config-volume/mount_path_/etc/coredns",
  "_toEntityKey": "default/12345/config-volume",
  "_type": "kube_container_spec_uses_volume",
  "displayName": "USES",
  "mountPath": "/etc/coredns",
  "mountPropagation": undefined,
  "readOnly": true,
  "subPath": undefined,
}
`;

exports[`#createContainerSpecToVolumeRelationship should generate relationship with mountPath and subPath 1`] = `
{
  "_class": "USES",
  "_fromEntityKey": "12345/coredns",
  "_key": "12345/coredns|uses|default/12345/config-volume/mount_config-volume/mount_path_/etc/coredns/mount_sub_path_abc",
  "_toEntityKey": "default/12345/config-volume",
  "_type": "kube_container_spec_uses_volume",
  "displayName": "USES",
  "mountPath": "/etc/coredns",
  "mountPropagation": undefined,
  "readOnly": true,
  "subPath": "abc",
}
`;

exports[`#createContainerSpecToVolumeRelationship should generate relationship without mountPath 1`] = `
{
  "_class": "USES",
  "_fromEntityKey": "12345/coredns",
  "_key": "12345/coredns|uses|default/12345/config-volume/mount_config-volume",
  "_toEntityKey": "default/12345/config-volume",
  "_type": "kube_container_spec_uses_volume",
  "displayName": "USES",
  "mountPath": undefined,
  "mountPropagation": undefined,
  "readOnly": true,
  "subPath": undefined,
}
`;

exports[`#createDeploymentEntity should convert data 1`] = `
{
  "_class": [
    "Deployment",
  ],
  "_key": "b39ade4b-d8f4-44a9-adc2-dff35d05514c",
  "_rawData": [
    {
      "name": "default",
      "rawData": {
        "metadata": {
          "annotations": {
            "deployment.kubernetes.io/revision": "1",
          },
          "creationTimestamp": 2021-04-13T15:37:33.000Z,
          "generation": 1,
          "labels": {
            "app": "kubernetes-bootcamp",
          },
          "name": "kubernetes-bootcamp",
          "namespace": "default",
          "resourceVersion": "92421",
          "uid": "b39ade4b-d8f4-44a9-adc2-dff35d05514c",
        },
        "spec": {
          "progressDeadlineSeconds": 600,
          "replicas": 1,
          "revisionHistoryLimit": 10,
          "selector": {
            "matchLabels": {
              "app": "kubernetes-bootcamp",
            },
          },
          "template": {
            "metadata": {
              "creationTimestamp": undefined,
              "labels": {
                "app": "kubernetes-bootcamp",
              },
            },
            "spec": {
              "containers": [
                {
                  "image": "gcr.io/google-samples/kubernetes-bootcamp:v1",
                  "imagePullPolicy": "IfNotPresent",
                  "name": "kubernetes-bootcamp",
                  "resources": {},
                  "terminationMessagePath": "/dev/termination-log",
                  "terminationMessagePolicy": "File",
                },
              ],
              "dnsPolicy": "ClusterFirst",
              "restartPolicy": "Always",
              "schedulerName": "default-scheduler",
              "securityContext": {},
              "terminationGracePeriodSeconds": 30,
            },
          },
        },
        "status": {
          "availableReplicas": 1,
          "conditions": [
            {
              "lastTransitionTime": 2021-04-13T15:37:33.000Z,
              "lastUpdateTime": 2021-04-13T15:37:46.000Z,
              "message": "ReplicaSet "kubernetes-bootcamp-57978f5f5d" has successfully progressed.",
              "reason": "NewReplicaSetAvailable",
              "status": "True",
              "type": "Progressing",
            },
            {
              "lastTransitionTime": 2021-04-16T18:10:54.000Z,
              "lastUpdateTime": 2021-04-16T18:10:54.000Z,
              "message": "Deployment has minimum availability.",
              "reason": "MinimumReplicasAvailable",
              "status": "True",
              "type": "Available",
            },
          ],
          "observedGeneration": 1,
          "readyReplicas": 1,
          "replicas": 1,
          "updatedReplicas": 1,
        },
      },
    },
  ],
  "_type": "kube_deployment",
  "active": undefined,
  "createdOn": 1618328253000,
  "deletionGracePeriodSeconds": undefined,
  "displayName": "kubernetes-bootcamp",
  "generation": 1,
  "minReadySeconds": undefined,
  "name": "kubernetes-bootcamp",
  "paused": undefined,
  "progressDeadlineSeconds": 600,
  "replicas": 1,
  "resourceVersion": "92421",
  "revisionHistoryLimit": 10,
  "status.availableReplicas": 1,
  "status.collisionCount": undefined,
  "status.observedGeneration": 1,
  "status.readyReplicas": 1,
  "status.replicas": 1,
  "status.unavailableReplicas": undefined,
  "status.updatedReplicas": 1,
  "strategy.maxSurge": undefined,
  "strategy.maxUnavailable": undefined,
  "strategy.type": undefined,
}
`;
